### PR TITLE
Update the JS Standard with es6 examples

### PR DESCRIPTION
**Pull request by jimmynotjim**
Yet another small change in preparation for further future standards updates.

## Changes

- Update all the code examples to use es6 variable and function formatting.

## Checklist

- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by sebworks** [comment]
:1

File: standards/javascript.md [hunk 1]
```diff
@@ -188,9 +187,9 @@ to share the same line of code.
 ##### Acceptable
 
 ```js
-var a = 'a';
-var b = 2;
-var i, j;
+const a = 'a';
+const b = 2;
+let i, j;
```
Comment (anselmbradford) [inline standards/javascript.md]: How would everyone feel about ditching this one? One variable declaration per line is clearer I think.

File: standards/javascript.md [hunk 2]
```diff
@@ -117,13 +117,13 @@ prefer to use single quotation marks `'`.
...
 ```js
-var message = 'oh hai ' + name + "!";
+const message = 'oh hai ' + name + "!";
 ```
 
 ##### Good
 
 ```js
-var message = 'oh hai ' + name + '!';
+const message = 'oh hai ' + name + '!';
```
Comment (anselmbradford) [inline standards/javascript.md]: Let's make this a template string `const message = `oh hai ${ name }!`;

Comment (Scotchester) [reply standards/javascript.md]: This is the section about single-quoting strings, though. We should probably have a separate section that addresses template strings.

File: standards/javascript.md [hunk 3]
```diff
@@ -141,41 +141,40 @@ same page.
 ##### Bad
 
 ```js
-var foo = 1,
+const foo = 1,
     bar = 2;
```
Comment (Scotchester) [inline standards/javascript.md]: I know this is the bad example, but it's demonstrating the inconsistency. Indent `bar` here to match `foo`; we're not animals.

File: standards/javascript.md (hunk 2 above)
Comment (anselmbradford) [reply standards/javascript.md]: That sounds better, and remove the concatenation here perhaps. It's just about the consistent quoting.

File: standards/javascript.md (hunk 1 above)
Comment (Scotchester) [reply standards/javascript.md]: Yeah, I think we should get more strict about requiring one per line. Also applies to L147.

File: standards/javascript.md (hunk 2 above)
Comment (jimmynotjim) [reply standards/javascript.md]: I didn't update any of the guide content in this PR but I have another branch that makes exactly this change and expands on the examples a bit.

File: standards/javascript.md (hunk 1 above)
Comment (jimmynotjim) [reply standards/javascript.md]: I thought this was weird too, and considered updating it in the other branch I have going. I'll update it there and we can discuss in the code.

File: standards/javascript.md [hunk 4]
```diff
@@ -285,7 +284,7 @@ instead of [function expressions][18]. Because [hoisting][19].
 ##### Bad
 
 ```js
-var sum = function( x, y ) {
+const sum = ( x, y ) => {
```
Comment (Scotchester) [inline standards/javascript.md]: Don't we encourage arrow functions in other scenarios? If so, I don't want people to misinterpret this as all arrow functions are bad.

File: standards/javascript.md (hunk 2 above)
Comment (anselmbradford) [reply standards/javascript.md]: 👍

File: standards/javascript.md (hunk 4 above)
Comment (jimmynotjim) [reply standards/javascript.md]: Just ones that will be hoisted like it says in the content, has nothing to do with it being an arrow or not an arrow.

File: standards/javascript.md (hunk 3 above)
Comment (jimmynotjim) [reply standards/javascript.md]: See the content between the two code blocks.

File: standards/javascript.md (hunk 4 above)
Comment (Scotchester) [reply standards/javascript.md]: Right, I just worry that dummies like me would see the arrow here, but not in the bad example, and think that the arrow was part of what a "function expression is" and assume they're always bad.

File: standards/javascript.md (hunk 3 above)
Comment (Scotchester) [reply standards/javascript.md]: Sorry, I wrote that poorly. Let me try again:

`bar` was aligned with `foo` before this change. It should still be.

Comment (jimmynotjim) [reply standards/javascript.md]: Ohhh, gotcha, I'll fix that in the next branch
